aws/resource_aws_licensemanager_association.go

```diff
@@ -0,0 +1,146 @@
+			"resource_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"license_configuration_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsLicenseManagerAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).licensemanagerconn
+
+	resourceArn := d.Get("resource_arn").(string)
+	licenseConfigurationArn := d.Get("license_configuration_arn").(string)
+
+	opts := &licensemanager.UpdateLicenseSpecificationsForResourceInput{
+		AddLicenseSpecifications: []*licensemanager.LicenseSpecification{{
+			LicenseConfigurationArn: aws.String(licenseConfigurationArn),
+		}},
+		ResourceArn: aws.String(resourceArn),
+	}
+
+	log.Printf("[DEBUG] License Manager association: %s", opts)
+
+	_, err := conn.UpdateLicenseSpecificationsForResource(opts)
+	if err != nil {
+		return fmt.Errorf("Error creating License Manager association: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s,%s", resourceArn, licenseConfigurationArn))
+
+	return resourceAwsLicenseManagerAssociationRead(d, meta)
+}
+
+func resourceAwsLicenseManagerAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).licensemanagerconn
+
+	resourceArn, licenseConfigurationArn, err := parseLicenseManagerAssociationId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	licenseSpecification, err := findLicenseSpecificationForResource(conn, resourceArn, licenseConfigurationArn)
+	if err != nil {
+		return fmt.Errorf("Error reading License Manager association: %s", err)
+	}
+
+	if licenseSpecification == nil {
+		log.Printf("[WARN] License Manager association (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("resource_arn", resourceArn)
+	d.Set("license_configuration_arn", licenseConfigurationArn)
+
+	return nil
+}
+
+func findLicenseSpecificationForResource(conn *licensemanager.LicenseManager, resourceArn, licenseConfigurationArn string) (*licensemanager.LicenseSpecification, error) {
+	opts := &licensemanager.ListLicenseSpecificationsForResourceInput{
+		ResourceArn: aws.String(resourceArn),
+	}
+
+	for {
+		resp, err := conn.ListLicenseSpecificationsForResource(opts)
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, licenseSpecification := range resp.LicenseSpecifications {
+			if aws.StringValue(licenseSpecification.LicenseConfigurationArn) == licenseConfigurationArn {
+				return licenseSpecification, nil
+			}
+		}
+
+		if len(resp.LicenseSpecifications) == 0 || resp.NextToken == nil {
+			return nil, nil
+		}
+
+		opts.NextToken = resp.NextToken
+	}
+}
+
+func parseLicenseManagerAssociationId(id string) (string, string, error) {
+	parts := strings.SplitN(id, ",", 2)
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("Could not parse License Manager association ID: %s", id)
```

Comment:
Nit: Might be helpful to show the expected ID format for import:

```suggestion
		return "", "", fmt.Errorf("Expected License Manager Association ID in format RESOURCEARN,LICENSEARN, received: %s", id)
```